pandas/_libs/lib.pyx

```diff
@@ -651,6 +651,11 @@ cpdef ndarray[object] ensure_string_array(
     cdef:
         Py_ssize_t i = 0, n = len(arr)
 
+    if hasattr(arr, "to_numpy"):
+        arr = arr.to_numpy()
+    elif not isinstance(arr, np.ndarray):
+        arr = np.array(arr, dtype="object")
```

Comment:
this should probably be `asarray`

not a big deal, but my preference would have been to type the input arr as an ndarray and handle non-ndarray in the calling function